src/Core/Domain/Product/VirtualProductFile/Command/AddVirtualProductFileCommand.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\VirtualProductFile\Command;
+
+use DateTimeInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+
+/**
+ * Adds downloadable file for virtual product
+ *
+ * @todo: is VirtualProductFile name correct for Core/Domain? or should I use ProductDownload everywhere
+ */
+class AddVirtualProductFileCommand
+{
+    /**
+     * @var ProductId
+     */
+    private $productId;
+
+    /**
+     * @var string
+     */
+    private $filePath;
+
+    /**
+     * @var string
+     */
+    private $displayName;
+
+    /**
+     * @var DateTimeInterface|null
+     */
+    private $expirationDate;
+
+    /**
+     * @var int|null
+     */
+    private $accessDays;
+
+    /**
+     * @var int|null
+     */
+    private $downloadTimesLimit;
+
+    /**
+     * @param int $productId
+     * @param string $filePath
+     * @param string $displayName display name of the file
+     * @param int|null $accessDays
+     * @param int|null $downloadTimesLimit
+     * @param DateTimeInterface|null $expirationDate
+     */
+    public function __construct(
+        int $productId,
+        string $filePath,
+        string $displayName,
+        ?int $accessDays = null,
+        ?int $downloadTimesLimit = null,
+        ?DateTimeInterface $expirationDate = null
+        //@todo: is_shareable & active? - there is no input in Product page for that. Is it still usable?
```

Comment:
I couldn't any usage of `is_shareable`
But `active` seems to be used when product is updated and its type is modified, when it's no longer a virtual product the ProductDownload is unactivated I guess it's a kind of soft delete which allows to re-enable it easily (can be useful in case of a false manipulation in the BO)

Reply:
I've checked actually, when product is not virtual anymore, the ProductDownload is just deleted.
It is also instantly deleted once you click **NO** here:
![Screenshot from 2020-11-10 09-45-11](https://user-images.githubusercontent.com/31609858/98644694-8c5dd600-2339-11eb-878c-8071698478c5.png)

Reply:
So I find those unused (at least in Add process). I remove the todo.